{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[]},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"4QWBCR-DpI-y","executionInfo":{"status":"ok","timestamp":1649216692530,"user_tz":-360,"elapsed":36932,"user":{"displayName":"Al Shahreyaj","userId":"07957919151863977060"}},"outputId":"4957480b-2ada-46ba-c977-1c87d4512ab7"},"outputs":[{"output_type":"stream","name":"stdout","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n","^C\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"code","source":["!cp -R '/content/drive/MyDrive/Colab Notebooks/Project/noise-reducer-ml/Dataset' /content/test"],"metadata":{"id":"MNBKeL_e-n3a"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["%cd /content/drive/MyDrive/Colab Notebooks/Project/noise-reducer-ml/denoiser-inverse/"],"metadata":{"id":"huH7COiStW8o","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1649216172783,"user_tz":-360,"elapsed":519,"user":{"displayName":"Al Shahreyaj","userId":"07957919151863977060"}},"outputId":"ed7895aa-7a5c-472a-d963-2e03f5db8844"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/Colab Notebooks/Project/noise-reducer-ml/denoiser-inverse\n"]}]},{"cell_type":"code","source":["!python3 convert.py -ni '/content/test/noisy.json' -ci '/content/test/clean.json' -o '/content/test/output' --num-shards-train 1 --num-shards-test 1 --num-shards-val 1 --duration 4 --sample-rate 48000 --test-size 0.2 --val-size 0.2"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"_eRLyL9Q0yf0","executionInfo":{"status":"ok","timestamp":1649216484442,"user_tz":-360,"elapsed":5007,"user":{"displayName":"Al Shahreyaj","userId":"07957919151863977060"}},"outputId":"0543a8c6-cb8c-43c0-f482-192dae582df0"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Splitting train set into TFRecord shards...\n","[('/content/test/output/train-001-6.tfrecord', array([0, 1, 2, 3, 4, 5]))]\n","Splitting test set into TFRecord shards...\n","[('/content/test/output/train-001-6.tfrecord', array([0, 1, 2, 3, 4, 5])), ('/content/test/output/test-001-3.tfrecord', array([6, 7, 8]))]\n","Splitting validate set into TFRecord shards...\n","[('/content/test/output/train-001-6.tfrecord', array([0, 1, 2, 3, 4, 5])), ('/content/test/output/test-001-3.tfrecord', array([6, 7, 8])), ('/content/test/output/validate-001-3.tfrecord', array([ 9, 10, 11]))]\n","\r  0% 0/3 [00:00<?, ?it/s]2022-04-06 03:41:23.029699: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected\n","2022-04-06 03:41:23.034159: W tensorflow/core/framework/op_kernel.cc:1745] OP_REQUIRES failed at whole_file_read_ops.cc:116 : NOT_FOUND: /content/test/Dataset/noisy/Copy of p287_001.wav; No such file or directory\n","2022-04-06 03:41:23.041543: W tensorflow/core/framework/op_kernel.cc:1745] OP_REQUIRES failed at whole_file_read_ops.cc:116 : NOT_FOUND: /content/test/Dataset/noisy/p287_001.wav; No such file or directory\n","\r  0% 0/3 [00:00<?, ?it/s]\n","multiprocessing.pool.RemoteTraceback: \n","\"\"\"\n","Traceback (most recent call last):\n","  File \"/usr/lib/python3.7/multiprocessing/pool.py\", line 121, in worker\n","    result = (True, func(*args, **kwds))\n","  File \"convert.py\", line 157, in _write_tfrecord_file\n","    noisy_audio = tf.io.read_file(n_file_path)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/io_ops.py\", line 133, in read_file\n","    return gen_io_ops.read_file(filename, name)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/gen_io_ops.py\", line 567, in read_file\n","    filename, name=name, ctx=_ctx)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/gen_io_ops.py\", line 590, in read_file_eager_fallback\n","    attrs=_attrs, ctx=ctx, name=name)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/execute.py\", line 55, in quick_execute\n","    inputs, attrs, num_outputs)\n","tensorflow.python.framework.errors_impl.NotFoundError: /content/test/Dataset/noisy/Copy of p287_001.wav; No such file or directory [Op:ReadFile]\n","\"\"\"\n","\n","The above exception was the direct cause of the following exception:\n","\n","Traceback (most recent call last):\n","  File \"convert.py\", line 328, in <module>\n","    main(parse_args())\n","  File \"convert.py\", line 321, in main\n","    converter.convert()\n","  File \"convert.py\", line 246, in convert\n","    _parallelize(self._write_tfrecord_file, shard_splits)\n","  File \"convert.py\", line 44, in _parallelize\n","    list(tqdm(pool.imap_unordered(func, data), total=len(data)))\n","  File \"/usr/local/lib/python3.7/dist-packages/tqdm/std.py\", line 1195, in __iter__\n","    for obj in iterable:\n","  File \"/usr/lib/python3.7/multiprocessing/pool.py\", line 748, in next\n","    raise value\n","  File \"/usr/lib/python3.7/multiprocessing/pool.py\", line 121, in worker\n","    result = (True, func(*args, **kwds))\n","  File \"convert.py\", line 157, in _write_tfrecord_file\n","    noisy_audio = tf.io.read_file(n_file_path)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/io_ops.py\", line 133, in read_file\n","    return gen_io_ops.read_file(filename, name)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/gen_io_ops.py\", line 567, in read_file\n","    filename, name=name, ctx=_ctx)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/gen_io_ops.py\", line 590, in read_file_eager_fallback\n","    attrs=_attrs, ctx=ctx, name=name)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/execute.py\", line 55, in quick_execute\n","    inputs, attrs, num_outputs)\n","tensorflow.python.framework.errors_impl.NotFoundError: /content/test/Dataset/noisy/Copy of p287_001.wav; No such file or directory [Op:ReadFile]\n"]}]},{"cell_type":"code","source":["rm -r *"],"metadata":{"id":"kBGdEgR79zvq"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":[],"metadata":{"id":"rrxo5QZ5-eQf"},"execution_count":null,"outputs":[]}]}